service.go

```diff
@@ -155,19 +155,19 @@ func (service *MgoService) newSession() *mgo.Session {
 
 // Stop stops the service.
 func (service *MgoService) Stop() error {
-	if service.running {
+	if service.isRunning() {
 		service.session.Close()
-		service.running = false
+		service.setRunning(false)
 	}
 	return nil
 }
 
 // RunWithSession runs a handler passing a new instance of the a session.
 func (service *MgoService) RunWithSession(handler func(session *mgo.Session) error) error {
-	if !service.running {
-		return rscsrv.ErrServiceNotRunning
+	if service.isRunning() {
```

Comment:
I'd prefer the previous writing with [the guard cause instead of nesting the main flow in a conditional](https://sourcemaking.com/refactoring/replace-nested-conditional-with-guard-clauses).

Reply:
I also agree with this approach (Design Pattern).